dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/ToolMetadata.java

```diff
@@ -22,32 +22,32 @@
 /**
  *
  */
-public record ToolMetadata(Tool annotation, Bean<?> bean, AnnotatedMethod<?> method, Map<String, ArgumentMetadata> arguments, String name, String title, String description) {
+
+public record ToolMetadata(Tool.Annotations annotations, Tool annotation, Bean<?> bean, AnnotatedMethod<?> method,
+                           Map<String, ArgumentMetadata> arguments) {
 
     public record ArgumentMetadata(Type type, int index, String description) {}
 
     public static ToolMetadata createFrom(Tool annotation, Bean<?> bean, AnnotatedMethod<?> method) {
+        return new ToolMetadata(
+                                annotation.annotations(),
+                                annotation,
+                                bean,
+                                method,
+                                getArgumentMap(method));
+    }
 
-        String name = annotation.name();
-        String title = annotation.title();
-        String description = annotation.description();
-
-        if (name.equals(Tool.ELEMENT_NAME)) {
-            name = method.getJavaMember().getName();
-        }
-        if (title.equals("")) {
-            title = null;
-        }
-
-        return new ToolMetadata(annotation, bean, method, getArgumentMap(method), name, title, description);
+    public ToolMetadata(Tool annotation, Bean<?> bean, AnnotatedMethod<?> method) {
+        this(annotation.annotations(), annotation, bean, method, getArgumentMap(method));
```

Comment:
You can't remove the `name`, `title` and `description` fields here. They don't always match the annotation.

Reply:
Removing this should have broken some tests. I've reopened #32372 to get the missing tests added.